01-Login/Dockerfile

```diff
@@ -1,11 +1,36 @@
-FROM node:8.7-alpine
+FROM node:10-alpine as build
 
-WORKDIR /home/app
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash git openssh
+
+RUN mkdir /app
+
+WORKDIR /app
+
+COPY package.json .
 
-ADD package.json /home/app
 RUN npm install
-ADD . /home/app
 
-CMD ["npm", "start"]
+COPY . .
+
+RUN npm run build
+
+# ---------------
+
+FROM node:10-alpine
+
+RUN mkdir -p /app/dist
+
+WORKDIR /app
+
+COPY --from=build /app/dist ./dist
```

Comment:
So this is a two-stage Docker build:

* The first stage builds the Vue app using Vue CLI 3 through `npm run build`, which outputs the HTML, JavaScript and CSS into the `./dist` folder. To do this, all the dependencies need to be available
* The second stage is a much slimmer build just for the runtime; it doesn't need all of the dev dependencies, it just need to spin up the Express server and serve an HTML page + assets

The result is that you end up with a Docker image that is a fraction of the size it would normally have been. I could probably get it even smaller using an Nginx-based image if that's something that's really important to us.